src/apis/logs_api.ts

```diff
@@ -1,25 +1,116 @@
-        ABIDecoder.addABI(DebtKernel.abi);
-        ABIDecoder.addABI(RepaymentRouter.abi);
+    /**
+     * @example
+     * // Get all "LogLoanOrderFilled" events between blocks 0 and 300.
+     * await dharma.logs.get(["LogLoanOrderFilled"], { from: 0, to: 300 });
+     *
+     * // Get the last 10 "LogLoanOrderFilled" events between blocks 0 and 300.
+     * await dharma.logs.get(["LogLoanOrderFilled"], { from: 0, to: 300, limit: 10 });
+     *
+     * // Get the last 10 "LogLoanOrderFilled" events.
+     * await dharma.logs.get(["LogLoanOrderFilled"], { limit: 10 });
+     *
+     * // Get the last 10 "LogLoanOrderFilled" or "LogLoanOrderCancelled" events that occurred.
+     * await dharma.logs.get(["LogLoanOrderFilled", "LogLoanOrderCancelled"], { limit: 10 });
+     *
+     * // Get all of the "LogLoanOrderFilled" events.
+     * await dharma.logs.get(["LogLoanOrderFilled"]);
+     *
+     * // Get all of the "LogLoanOrderFilled" events.
+     * await dharma.logs.get("LogLoanOrderFilled");
+     *
+     * @param {string | string[]} eventNames
+     * @param {GetEventOptions} options
+     * @returns {Promise<any>}
+     */
+    public async get(eventNames: string | string[], options: GetEventOptions = {}): Promise<any> {
+        const { fromBlock, limit, toBlock } = options;
+
+        eventNames = _.castArray(eventNames);
+
+        const eventToContract = await this.getEventToContractsMap();
+
+        let events = [];
+
+        await Promise.all(
+            eventNames.map(async (eventName) => {
+                const contractWrapper = eventToContract[eventName];
+                const contract = this.web3.eth
+                    .contract(contractWrapper.abi)
+                    .at(contractWrapper.address);
+
+                return new Promise((resolve) => {
+                    const eventsWatcher = contract.allEvents({
+                        fromBlock: fromBlock || 0,
+                        toBlock: toBlock || "latest",
+                    });
+
+                    eventsWatcher.get((error, logs) => {
+                        const filteredEvents = _.filter(logs, (log) => log.event === eventName);
+
+                        events = events.concat(filteredEvents);
+
+                        resolve();
+                    });
+
+                    eventsWatcher.stopWatching();
+                });
+            }),
+        );
+
+        if (_.isNumber(limit)) {
+            return _.take(events, limit);
+        } else {
+            return events;
+        }
+    }
+
+    /**
+     * Returns a list of events that are obtainable from Dharma Protocol contracts.
+     *
+     * @example
+     * dharma.logs.list()
+     * => [
+     *  "LogDebtOrderFilled",
+     *  "LogIssuanceCancelled",
+     *  "LogDebtOrderCancelled",
+     *  "LogError",
+     *  "Pause",
+     *  ...
+     * ]
+     *
+     * @returns {string[]}
+     */
+    public async list(): Promise<string[]> {
```

Comment:
do we need to expose this publicly?

Reply:
Yeah I think so, as a way to know all of the events they can listen to. Should also be in the docs.

Reply:
feels weird to me that they'd want to programmatically want this as opposed to looking it up in our documentation.

Reply:
Yeah but what if you wanted to listen to all of the events? you could just do:
```
const logTypes = dharma.logs.limit();
const logs = dharma.logs.get(logTypes);
```

Reply:
easier thing would be to do this for them via an endpoint:

`listenToAllEvents()`